refined object storage migration assembly (#638)
openstack-k8s-operators/data-plane-adoption: docs_user/assemblies/assembly_migrating-the-object-storage-service.adoc

:context: migrate-object-storage-service

= Migrating the {object_storage} to {rhos_long_noacro} nodes
If you are using the {rhos_prev_long} {object_storage_first_ref} as an Object Storage service, you must migrate your Object Storage service to {rhos_long_noacro} nodes. If you are using the Object Storage API of the Ceph Object Gateway (RGW), you can skip this chapter.

> The data migration happens replica by replica. For example, if you start with 3 replicas, you only move 1 one of them at a time to ensure that the remaining 2 replicas are still available and that you can use the {object_storage} during the migration.

Reviewer: ```suggestion
The data migration happens replica by replica. For example, if you have 3 replicas, move them one at a time to ensure that the other 2 replicas are still operational, which enables you to continue to use the {object_storage} during the migration.
```
A couple of suggestions to help with clarity around the what and the why.

[NOTE]

> Data migration to the new deployment is a long-running process that runs mostly in the background. The {object_storage} replicators move data from old to new nodes, but depending on the amount of used storage, this might take a long time. To reduce downtime, you can use the old nodes if they are running and continue with adopting other services in the meantime. Performance might be decreased to the amount of replication traffic in the network.

Author: 1. Can we give an estimate of how long it might take to move data from old to new nodes?
2. Regarding the last sentence in the note, is that supposed to say "Performance might be degraded **due to** the mount of replication traffic in the network"?

Reviewer 2: 1. Unfortunately not - can be minutes up to months. It depends mostly on the amount of storage, but also on disk speed, network speed, overall load etc.
2. Yes

> Data migration to the new deployment is a long-running process that runs mostly in the background. The {object_storage} replicators move data from old to new nodes, but depending on the amount of used storage, this might take a long time. To reduce downtime, you can use the old nodes if they are running and continue with adopting other services in the meantime. Performance might be degraded due to the amount of replication traffic in the network.

Reviewer: ```suggestion
Data migration to the new deployment is a long-running process that executes mostly in the background. The {object_storage} replicators move data from old to new nodes, which might take a long time depending on the amount of storage used. To reduce downtime, you can use the old nodes if they are running and continue with adopting other services while waiting for the migration to complete. Performance might be degraded due to the amount of replication traffic in the network.
```
Couple of suggestions, none deal-breakers. A couple for clarity and ease of parsing.
